src/index.js

```diff
@@ -1,7 +1,102 @@
-export default function main (input, options = {}) {
-  if (typeof input !== 'string') {
-    throw new TypeError('Expected input to be string');
+const commandExists = require('command-exists').sync;
+const debug = require('debug')('ns-audio:image');
+const ffmpeg = require('fluent-ffmpeg');
+const fs = require('fs');
+const fsPromises = require('fs').promises;
+const gnuplot = commandExists('gnuplot') ? 'gnuplot' : 'gnuplot-nox';
+const pEvent = require('p-event');
+const spawn = require('child_process').spawn;
+const tempy = require('tempy');
+
+// Handle ffmpeg-static and ffprobe static optional deps.
+let ffmpegPath;
+let ffprobePath;
+try {
+  ffmpegPath = require('ffmpeg-static');
+} catch {
+  ffmpegPath = 'ffmpeg';
+}
+
+try {
+  ffprobePath = require('ffprobe-static').path;
+} catch {
+  ffprobePath = 'ffprobe';
+}
+
+ffmpeg.setFfmpegPath(ffmpegPath);
+ffmpeg.setFfprobePath(ffprobePath);
+
+function _getDuration (filename) {
+  return new Promise((resolve, reject) => {
+    ffmpeg.ffprobe(filename, (err, metadata) => {
+      if (err) return reject(err);
+      const duration = metadata.streams && metadata.streams.length && Number.parseFloat(metadata.streams[0].duration);
+      if (Number.isNaN(duration)) {
```

Comment:
be careful using `Number.isNaN()`. It doesn't quite do what you think it is doing...
```js
isNaN(undefined); // true
Number.isNaN(undefined); // false
```
ie. `Number.isNaN()` is checking specifically whether the value is `NaN` only and not whether when it is converted to a number it is `NaN`. To use `Number.isNaN()` you need to do something like
```js
Number.isNaN(Number(undefined)); // true
```